kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriptionServiceImpl.java

```diff
@@ -390,51 +305,71 @@ private Subscription createSubscriptionViaIvr(Subscriber subscriber, Subscriptio
      *    subscription to Pregnancy Pack on this MSISDN.
      */
     private Subscription createSubscriptionViaMcts(Subscriber subscriber, SubscriptionPack pack) {
-        Subscription subscription;
 
-        if (subscriber.getDateOfBirth() != null && pack.getType() == SubscriptionPackType.CHILD) {
-            // DOB (with or without LMP) is present
-            if (getActiveSubscription(subscriber, SubscriptionPackType.CHILD) != null)
-            {
+        if (!enrollmentPreconditionCheck(subscriber, pack)) {
+            return null;
+        }
+
+        DateTime startDate = (pack.getType() == SubscriptionPackType.CHILD) ?
+                subscriber.getDateOfBirth() : subscriber.getLastMenstrualPeriod().plusDays(KilkariConstants.THREE_MONTHS);
+
+        Subscription subscription = new Subscription(subscriber, pack, SubscriptionOrigin.MCTS_IMPORT);
+        subscription.setStartDate(startDate);
+        subscription.setNeedsWelcomeMessageViaObd(true);
+
+        if (allowMctsSubscriptions) {
```

Comment:
Don't you wanna put that check at the beginning of the method that way you don't even have to call enrollmentPreconditionCheck?